diff --git a/sql/handler.cc b/sql/handler.cc
index 6d64ab4..096c9e5 100644
--- a/sql/handler.cc
+++ b/sql/handler.cc
@@ -5638,31 +5638,14 @@ bool default_rm_tmp_tables(handlerton *hton, THD *, List<LEX_STRING> *files) {
 }
 
 /*****************************************************************************
-  Key cache handling.
-
-  This code is only relevant for ISAM/MyISAM tables
-
-  key_cache->cache may be 0 only in the case where a key cache is not
-  initialized or when we where not able to init the key cache in a previous
-  call to ha_init_key_cache() (probably out of memory)
+  Legacy key cache handling. InnoDB does not use the MyISAM key cache.
 *****************************************************************************/
 
 /**
   Init a key cache if it has not been initied before.
 */
 int ha_init_key_cache(std::string_view, KEY_CACHE *key_cache) {
-  DBUG_TRACE;
-
-  if (!key_cache->key_cache_inited) {
-    mysql_mutex_lock(&LOCK_global_system_variables);
-    const size_t tmp_buff_size = (size_t)key_cache->param_buff_size;
-    const ulonglong tmp_block_size = key_cache->param_block_size;
-    const ulonglong division_limit = key_cache->param_division_limit;
-    const ulonglong age_threshold = key_cache->param_age_threshold;
-    mysql_mutex_unlock(&LOCK_global_system_variables);
-    return !init_key_cache(key_cache, tmp_block_size, tmp_buff_size,
-                           division_limit, age_threshold);
-  }
+  (void)key_cache;
   return 0;
 }
 
@@ -5670,20 +5653,7 @@ int ha_init_key_cache(std::string_view, KEY_CACHE *key_cache) {
   Resize key cache.
 */
 int ha_resize_key_cache(KEY_CACHE *key_cache) {
-  DBUG_TRACE;
-
-  if (key_cache->key_cache_inited) {
-    mysql_mutex_lock(&LOCK_global_system_variables);
-    const size_t tmp_buff_size = (size_t)key_cache->param_buff_size;
-    const ulonglong tmp_block_size = key_cache->param_block_size;
-    const ulonglong division_limit = key_cache->param_division_limit;
-    const ulonglong age_threshold = key_cache->param_age_threshold;
-    mysql_mutex_unlock(&LOCK_global_system_variables);
-    const int retval =
-        resize_key_cache(key_cache, keycache_thread_var(), tmp_block_size,
-                         tmp_buff_size, division_limit, age_threshold);
-    return !retval;
-  }
+  (void)key_cache;
   return 0;
 }
 

